.github/workflows/android.yml

```diff
@@ -17,134 +17,121 @@
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-          submodules: true
+      - uses: actions/checkout@v3
 
-      - name: Clone SDL Android repo
-        run: |
-          git clone --depth=100 https://github.com/pelya/commandergenius.git build.android
-          git -C build.android checkout ${SDL_ANDROID_REV}
+      - name: Set script permissions
+        run: chmod +x ./.ci_scripts/*.sh
 
-      - name: Clone SDL Android repo submodules
-        # There are many more submodules in this repo, we don't need them all
-        working-directory: build.android
-        run: >
-          git submodule update --init --recursive --depth=1
-          project/jni/boost/src project/jni/iconv/src
-          project/jni/sdl2 project/jni/sdl2_image
-          project/jni/sdl2_mixer project/jni/sdl2_ttf
+      - name: Use cached dependencies
+        id: use-dependencies
+        uses: actions/cache/restore@v3
+        with:
+          path: ./build.android/app/jni/
+          key: Android-${{ matrix.arch }}-dependencies-${{ hashFiles('.github/workflows/android.yml') }}
 
-      - name: Symlink the application dir
+      - name: Prepare SDL Android project
+        if: steps.use-dependencies.outputs.cache-hit != 'true'
         run: |
-          rm -rf build.android/project/jni/application/supertux/supertux
-          ln -s `pwd` build.android/project/jni/application/supertux/supertux
-          ln -s supertux build.android/project/jni/application/src
+          ./.ci_scripts/clone_dependency.sh libsdl-org/SDL release-2.26.5
+          mv ./SDL/android-project/ ./build.android
+          mv ./SDL/ ./build.android/app/jni/SDL
 
-      - name: Set architectures list and version
-        working-directory: build.android
+      - name: Clone dependencies
+        working-directory: build.android/app/jni
+        if: steps.use-dependencies.outputs.cache-hit != 'true'
         run: |
-          VERSION_NAME="$(git -C .. describe --tags | cut -f 1,2 -d "-" | sed 's/[^0-9\\.\\-]//g' | sed 's/-/./g')"
-          VERSION_CODE="$(printf "%02d%02d%02d%04d\n" "$(echo $VERSION_NAME | cut -d . -f 1)" "$(echo $VERSION_NAME | cut -d . -f 2)" "$(echo $VERSION_NAME | cut -d . -f 3)" "$(echo $VERSION_NAME | cut -d . -f 4)")"
-          echo "Version name: $VERSION_NAME"
-          echo "Version code: $VERSION_CODE"
-          sed -i "s/MultiABI=.*/MultiABI='${ARCH_LIST}'/g" project/jni/application/supertux/AndroidAppSettings.cfg
-          sed -i "s/AppVersionName.*/AppVersionName="$VERSION_NAME"/g" project/jni/application/supertux/AndroidAppSettings.cfg
-          sed -i "s/AppVersionCode.*/AppVersionCode=$VERSION_CODE/g" project/jni/application/supertux/AndroidAppSettings.cfg
-
-      - name: Copy precompiled libraries from cache
-        uses: actions/cache@v2
-        id: cache
+          ../../../.ci_scripts/clone_dependency.sh libsdl-org/SDL_image release-2.6.3
```

Comment:
Maybe work with pushd / popd here?

Reply:
the script clones into current directory

Reply:
Ah, okay.